### PR TITLE
Fixes Autoloader::loadLegacy() always called

### PR DESCRIPTION
**Pull request by samsonasik**
The `loadLegacy()` function was always called in `Autoloader::loadClass()` function for just normal `Home` Controller accessing `/` page, while there is no legacy class. /cc @paulbalandan if you can verify.

Tested in MacOS.

**Checklist:**
- [x] Securely signed commits
- [x] Unit testing, with >80% coverage

### CODE REVIEW

**Comment by samsonasik** [comment]
seems still not working, I will check more.

**Comment by samsonasik** [comment]
Ok, so, on debugging, got class `Modules` that fallback in `loadLegacy()`

**Comment by samsonasik** [comment]
It seems it happen on calling `config()` function, eg: `config('Modules')`

**Comment by samsonasik** [comment]
call `config('Modules)` -> loadLegacy() called.
call `config('Config\Modules)` -> loadLegacy() not called.

**Comment by samsonasik** [comment]
I added `Config\\` prefix when class provided is not namespaced. If it still not found in the `Config\\` namespace, then find `loadLegacy()` as fallback. By this, both `config('Modules)` won't call `loadLegacy()` function.

**Comment by samsonasik** [comment]
I've added unit test for it.

**Comment by samsonasik** [comment]
updated with faster find assumption of Config class without looping.

File: system/Autoloader/Autoloader.php [hunk 1]
```diff
@@ -280,6 +280,15 @@ protected function loadInNamespace(string $class)
 	{
 		if (strpos($class, '\\') === false)
 		{
+			$class    = 'Config\\' . $class;
+			$filePath = APPPATH . str_replace('\\', '/', $class) . '.php';
```
Comment (paulbalandan) [inline system/Autoloader/Autoloader.php:284]: Use `DIRECTORY_SEPARATOR` as replacement for pretty printing shutdown handler backtrace.

Comment (samsonasik) [reply system/Autoloader/Autoloader.php]: updated.

**Comment by MGatner** [comment]
@samsonasik Are all non-namespaced calls from `config()` triggering `loadLegacy()`? That might merit reworking.

**Comment by samsonasik** [comment]
travis build green 🎉

**Comment by samsonasik** [comment]
@MGatner Yes, it was down there if no namespace specified. Probably, the `loadlegacy()` can be removed for `4.1.0` ?

**Pull request merged**